staging/src/k8s.io/apiserver/pkg/authentication/request/x509/x509.go

```diff
@@ -176,6 +177,9 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 	*/
 
 	remaining := req.TLS.PeerCertificates[0].NotAfter.Sub(time.Now())
+	if remaining.Seconds() < 86400 {
+		klog.Warningf("Certificate of client: %s is about to expire in %f seconds", req.RemoteAddr, remaining.Seconds())
```

Comment:
Could you add more parameters to the warning message to make it clear what application is requesting it? For example req.UserAgent().

Also, the output in seconds is not very readable for humans—just my two cents.